pkg/kubernetes/renew_certificate.go

```diff
@@ -86,14 +88,23 @@ func parseCertificateFiles(rootCert, issuerCert, issuerKey string) ([]byte, []by
 	return rootCertBytes, issuerCertBytes, issuerKeyBytes, nil
 }
 
-func renewCertificate(rootCert, issuerCert, issuerKey []byte, timeout uint) error {
+func renewCertificate(rootCert, issuerCert, issuerKey []byte, timeout uint, imageVariant string) error {
+	var daprVersion, daprImageVariant string
 	status, err := GetDaprResourcesStatus()
 	if err != nil {
 		return err
 	}
-	daprVersion := GetDaprVersion(status)
+	daprVersion = GetDaprVersion(status)
 	print.InfoStatusEvent(os.Stdout, "Dapr control plane version %s detected in namespace %s", daprVersion, status[0].Namespace)
 
+	// Get the current version if image variant is provided.
+	if imageVariant != "" {
```

Comment:
`imageVariant` is just needed to extract the version from this string 1.x.x-mariner and do some validation on that. It is not needed in actual cert rotation logic as this will be handled by this logic - `upgradeClient.ReuseValues = true` below at line 118. It will reuse all previous flags if not passed.

Reply:
Can you add that as a comment in the code itself?

Reply:
Unrelated, but I believe image variants should not be passed around as strings. We should create a new type and use that instead.

Reply:
Agreed